javascript/ql/src/semmle/javascript/security/dataflow/FileAccessToHttp.qll

```diff
@@ -0,0 +1,87 @@
+/** 
+ * Provides Taint tracking configuration for reasoning about file access taint flow to html post body 
+ */
+import javascript
+import semmle.javascript.frameworks.HTTP
+
+module FileAccessToHttpDataFlow {
+  /**
+   * A data flow source for code injection vulnerabilities.
+   */
+  abstract class Source extends DataFlow::Node { }
+
+  /**
+   * A data flow sink for code injection vulnerabilities.
+   */
+  abstract class Sink extends DataFlow::Node { }
+
+  /**
+   * A sanitizer for code injection vulnerabilities.
+   */
+  abstract class Sanitizer extends DataFlow::Node { }
+
+  /**
+   * A taint-tracking configuration for reasoning about code injection vulnerabilities.
+   */
+  class Configuration extends TaintTracking::Configuration {
+    Configuration() { this = "FileAccessToHttpDataFlow" }
+
+    override predicate isSource(DataFlow::Node source) {
+      source instanceof Source
+    }
+
+    override predicate isSink(DataFlow::Node sink) {
+      sink instanceof Sink
+    }
+
+    override predicate isSanitizer(DataFlow::Node node) {
+      super.isSanitizer(node) or
+      node instanceof Sanitizer
+    }
+  }
+
+/** Taint step propagating a tainted parameter or a  variable to a wrapper object. */ 
+  private class ObjectTaintStep extends TaintTracking::AdditionalTaintStep {
+    DataFlow::Node target;
+    
+    ObjectTaintStep() {
+      (
+          this = DataFlow::valueNode(_) or
+          this = DataFlow::parameterNode(_) or
+          this instanceof DataFlow::PropRead
+      ) and
+      exists (DataFlow::PropWrite pwr |
+        target = pwr.getBase() and
+        this = pwr.getRhs()
+      )
+    }
+
+    override predicate step(DataFlow::Node pred, DataFlow::Node succ) {
+      pred = this and succ = target
+    }
+  }  
+
+  /** A source is a file access parameter, as in readFromFile(buffer). */
+  private class FileAccessArgumentAsSource extends Source {
+    FileAccessArgumentAsSource() {  
+      exists(DataFlow::CallNode cn |
+        cn instanceof FileSystemAccess and
+        this = cn.getAnArgument()
+      )
+    }
+  }
+
+  /** A source is a file access callback parameter, as in fs.readFile('MyFile.json', 'utf-8', function (err, data) {}. */
+  private class FileAccessParamAsSource extends Source {
+    FileAccessParamAsSource() {  
+      this instanceof FileSystemAccess
```

Comment:
This seems overly broad. Perhaps you'd want to restrict this to certain kinds of file system accesses?